### PR TITLE
Minor fix for DEFAULT constraint and refactor postgresparser

### PR DESCRIPTION
**Pull request by hzxa21**
This PR adds the missing part for parsing constant DEFAULT constraint #644 and refactor postgresparser by adding ExprTransform for all expression nodes for better modularity.

### CODE REVIEW

File: src/parser/postgresparser.cpp [hunk 1]
```diff
@@ -288,36 +288,23 @@ parser::GroupByDescription* PostgresParser::GroupByTransform(List* group,
   result->columns = new std::vector<expression::AbstractExpression*>();
   for (auto cell = group->head; cell != nullptr; cell = cell->next) {
     Node* temp = reinterpret_cast<Node*>(cell->data.ptr_value);
-    switch (temp->type) {
-      case T_ColumnRef:
```
Comment (luochenUmich) [inline src/parser/postgresparser.cpp]: Here, group by columns should only be either a tuple value expression or complex expressions like a + b. In your new code, other expressions will also be parsed and passed to the optimizer. Then, it becomes the optimizer's responsibility to throw the errors. But originally, the errors will come out of parser. Same problems for other places.

File: src/parser/postgresparser.cpp [hunk 2]
```diff
@@ -288,36 +288,23 @@ parser::GroupByDescription* PostgresParser::GroupByTransform(List* group,
...
-      default: {
-        delete result;
-        throw NotImplementedException(StringUtil::Format(
-          "HAVING of type %d not supported yet...", having->type));
-      }
+    try {
+      result->having = ExprTransform(having);
+    }
+    catch(NotImplementedException e) {
+      throw NotImplementedException(
```
Comment (luochenUmich) [inline src/parser/postgresparser.cpp]: Forget delete result before throwing exception?

File: src/parser/postgresparser.cpp [hunk 3]
```diff
@@ -1271,7 +1194,7 @@ parser::SQLStatementList* PostgresParser::ParseSQLString(
   }
 
   // DEBUG only. Comment this out in release mode
-//  print_pg_parse_tree(result.tree);
+  print_pg_parse_tree(result.tree);
```
Comment (luochenUmich) [inline src/parser/postgresparser.cpp]: Don't forget to comment this out

File: src/parser/postgresparser.cpp (hunk 1 above)
Comment (hzxa21) [inline src/parser/postgresparser.cpp]: You are right. GroupByTransform and OrderByTransform should not call the ExprTransform directly but in other cases it should be fine. Note that group by and order by can have constant expressions to directly indicate the column positions, which we may overlook in the new optimizer.

File: src/parser/postgresparser.cpp (hunk 2 above)
Comment (hzxa21) [inline src/parser/postgresparser.cpp]: Good catch.

File: src/parser/postgresparser.cpp (hunk 1 above)
Comment (hzxa21) [inline src/parser/postgresparser.cpp]: By the way, postgres documentation says that any expression is valid in group by column as long as it is formed from the input columns. I think only ParamExpressions is invalid in this case. See https://www.postgresql.org/docs/8.0/static/sql-select.html. I am also surprised that weird queries like `select a>1 from a group by a>1` also work in postgres. I guess the rewriter has some rules to eliminate group by.

**Pull request merged**